spec/unit/query_azure_mock.rb

```diff
@@ -7,10 +8,31 @@ def setup_query_azure_mock
     stub_query_azure (@connection)
   end
 
+  def mock_list_images
+    create_service
+    stub_list_images(@service)
+  end
+
   def create_connection
     @connection = Azure::Connection.new(TEST_PARAMS)
   end
 
+  def create_service
+    @service = Azure::ServiceManagement::ASMInterface.new(TEST_PARAMS)
+  end
+
+  def stub_list_images(service)
+    stub_query_azure(service.connection)
+  end
+
+  def stub_delete_server(service)
+    stub_query_azure(service.connection)
+  end
+
+  def stub_show_server(service)
+    stub_query_azure(service.connection)
+  end
```

Comment:
There is a repetition of `stub_query_azure(service.connection)` code inside `stub_list_images`, `stub_delete_server` and `stub_show_server` methods. We should be able to avoid this.